File: python/svix/api_internal/autoconfig_subscription.py
# This file is @generated
from ..models import AutoConfigSubscriptionOut
from .autoconfig_subscription_destination import (
    AutoconfigSubscriptionDestination,
    AutoconfigSubscriptionDestinationAsync,
)
from .autoconfig_subscription_endpoint import (
    AutoconfigSubscriptionEndpoint,
    AutoconfigSubscriptionEndpointAsync,
)
from .common import ApiBaseAsync, ApiBaseSync


class AutoconfigSubscriptionAsync(ApiBaseAsync):
    @property
    def destination(self) -> AutoconfigSubscriptionDestinationAsync:
        return AutoconfigSubscriptionDestinationAsync(self._client, self._httpx_client)

    @property
    def endpoint(self) -> AutoconfigSubscriptionEndpointAsync:
        return AutoconfigSubscriptionEndpointAsync(self._client, self._httpx_client)

    async def get(self, app_id: str, autoconfig_id: str) -> AutoConfigSubscriptionOut:
        """Get an AutoConfig subscription, including the bound endpoint or destination if any."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
            path_params={
                "app_id": app_id,
                "autoconfig_id": autoconfig_id,
            },
        )
        return AutoConfigSubscriptionOut.model_validate(response.json())


class AutoconfigSubscription(ApiBaseSync):
    @property
    def destination(self) -> AutoconfigSubscriptionDestination:
        return AutoconfigSubscriptionDestination(self._client, self._httpx_client)

    @property
    def endpoint(self) -> AutoconfigSubscriptionEndpoint:
        return AutoconfigSubscriptionEndpoint(self._client, self._httpx_client)

    def get(self, app_id: str, autoconfig_id: str) -> AutoConfigSubscriptionOut:
        """Get an AutoConfig subscription, including the bound endpoint or destination if any."""
        response = self._request_sync(
            method="get",
            path="/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
            path_params={
                "app_id": app_id,
                "autoconfig_id": autoconfig_id,
            },
        )
        return AutoConfigSubscriptionOut.model_validate(response.json())

File: csharp/Svix/ApiInternal/AutoconfigSubscription.cs
// this file is @generated
#nullable enable
using Microsoft.Extensions.Logging;
using Svix.Models;

namespace Svix.ApiInternal
{
    public class AutoconfigSubscription(SvixClient client)
    {
        readonly SvixClient _client = client;

        public AutoconfigSubscriptionDestination Destination
        {
            get => new AutoconfigSubscriptionDestination(_client);
        }

        public AutoconfigSubscriptionEndpoint Endpoint
        {
            get => new AutoconfigSubscriptionEndpoint(_client);
        }

        /// <summary>
        /// Get an AutoConfig subscription, including the bound endpoint or destination if any.
        /// </summary>
        public async Task<AutoConfigSubscriptionOut> GetAsync(
            string appId,
            string autoconfigId,
            CancellationToken cancellationToken = default
        )
        {
            try
            {
                var response =
                    await _client.SvixHttpClient.SendRequestAsync<AutoConfigSubscriptionOut>(
                        method: HttpMethod.Get,
                        path: "/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
                        pathParams: new Dictionary<string, string>
                        {
                            { "app_id", appId },
                            { "autoconfig_id", autoconfigId },
                        },
                        cancellationToken: cancellationToken
                    );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(GetAsync)} failed");

                throw;
            }
        }

        /// <summary>
        /// Get an AutoConfig subscription, including the bound endpoint or destination if any.
        /// </summary>
        public AutoConfigSubscriptionOut Get(string appId, string autoconfigId)
        {
            try
            {
                var response = _client.SvixHttpClient.SendRequest<AutoConfigSubscriptionOut>(
                    method: HttpMethod.Get,
                    path: "/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
                    pathParams: new Dictionary<string, string>
                    {
                        { "app_id", appId },
                        { "autoconfig_id", autoconfigId },
                    }
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(Get)} failed");

                throw;
            }
        }
    }
}
